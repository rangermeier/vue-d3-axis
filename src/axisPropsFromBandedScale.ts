import { BandedScale, PropsForAxis } from './types'

export function axisPropsFromBandedScale<T>(
  scale: BandedScale<T>,
): PropsForAxis<T> {
  const range = scale.range()
  const values = scale.domain()
  const format = (d: T): string => String(d)

  let offset = scale.bandwidth() / 2
  const scaleCopy = scale.copy()
  if (scaleCopy.round()) offset = Math.round(offset)
  const position = (d: T): number => scaleCopy(d) + offset

  return { range, values, format, position }
}
