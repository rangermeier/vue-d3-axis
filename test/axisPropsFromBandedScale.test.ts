import { scaleBand } from 'd3-scale'
import axisPropsFromBandedScale from '@/axisPropsFromBandedScale'

test('Works with bandwidth scales', () => {
  const scale = scaleBand()
    .domain(['foo', 'bar', 'baz'])
    .range([0, 500])
    .padding(1)

  const props = axisPropsFromBandedScale(scale)
  expect(props).toMatchSnapshot()

  expect(props.position('foo')).toEqual(125)
  expect(props.position('bar')).toEqual(250)
  expect(props.position('baz')).toEqual(375)

  expect(props.format('foo')).toEqual('foo')
})
