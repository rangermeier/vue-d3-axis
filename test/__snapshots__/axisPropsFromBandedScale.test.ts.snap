// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works with bandwidth scales 1`] = `
Object {
  "format": [Function],
  "position": [Function],
  "range": Array [
    0,
    500,
  ],
  "values": Array [
    "foo",
    "bar",
    "baz",
  ],
}
`;

exports[`test Works with bandwidth scales 1`] = `
Object {
  "format": [Function],
  "position": [Function],
  "range": Array [
    0,
    500,
  ],
  "values": Array [
    "foo",
    "bar",
    "baz",
  ],
}
`;
